tests/cases.py

```diff
@@ -108,3 +112,97 @@ def setUp(self):
     def tearDown(self) -> None:
         """Tear down the test case by cleaning up the temporary cache directory."""
         self.directory.cleanup()
+
+
+class LossTestCase(unittest.TestCase):
+    """Base unittest for loss functions."""
+
+    #: The class
```

Comment:
This common part could be extracted to a base class (similar than in #107 ). No blocker for this PR though,